tests/Export-DbaInstance.Tests.ps1

```diff
@@ -4,23 +4,334 @@ Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
 
 Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
     Context "Validate parameters" {
-        [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object {$_ -notin ('whatif', 'confirm')}
-        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Credential', 'Path', 'NoRecovery', 'IncludeDbMasterKey', 'Exclude', 'BatchSeparator', 'ScriptingOption', 'NoPrefix', 'ExcludePassword', 'Append', 'EnableException'
-        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
         It "Should only contain our specific parameters" {
-            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object {$_}) -DifferenceObject $params).Count ) | Should Be 0
+            [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object { $_ -notin ('whatif', 'confirm') }
+            [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Credential', 'Path', 'NoRecovery', 'IncludeDbMasterKey', 'Exclude', 'BatchSeparator', 'ScriptingOption', 'NoPrefix', 'ExcludePassword', 'Append', 'EnableException'
+            $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
```

Comment:
Is there a reason you are moving these into the It block on your PRs? I noticed you did this on the last PR you had submitted.

Reply:
Yes, with Pester 5.0.2 the following error occurs due to scoping of the variables. So, the fix I have been implementing is to just move the test code inside the It blocks so the test logic has the necessary scope visibility.

[-] Export-DbaInstance Unit Tests.Validate parameters.Should only contain our specific parameters 405ms (245ms|159ms)
 ParameterBindingValidationException: Cannot bind argument to parameter 'ReferenceObject' because it is null.
 at <ScriptBlock>, C:\Users\lancasteradam\Documents\GitHub\forks\dbatools\tests\Export-DbaInstance.Tests.ps1:11

Reply:
We are not updating to Pester 5 any time soon. When we do it will be to 5.1 because many things and structures around the discovery process that Jakub implemented changed and improved.

This can stay but please don't make these changes going forward on future PRs.

Our test will be written to support Pester 5+ as a whole so we handle applying the same pattern across the board.